src/ica_api.c

```diff
@@ -292,6 +292,16 @@ static unsigned int check_gcm_parms(unsigned long text_length,
 	return 0;
 }
 
+static unsigned int check_gcm_kma_parms(const kma_ctx* ctx, unsigned long data_length,
+									unsigned long aad_length)
+{
+	if (ctx->total_aad_length / 8 + aad_length > S390_GCM_MAX_AAD_LENGTH ||
```

Comment:
is the `/ 8` still correct here? Isn't it first added to `ctx->total_aad_length` in full bytes, and only right before the instruction is called is multiplied by 8 to turn it into number of bits? If so, then I guess you should not divide by 8 here.

Same for `total_input_length` of course.

Reply:
Hmm yes ... when using the "new" API, the lengths are already accumulated as bit values in routine s390_aes_gcm_kma:
```
	/* Actual lengths needed by KMA */
	ctx->total_aad_length += aad_length * 8;
	ctx->total_input_length += data_length * 8;

	/* Call KMA */
	rc = s390_kma(hw_fc, ctx,
			out_data, in_data, data_length,
			aad, aad_length);
```
The KMA instruction needs the values in bits. So we have a discrepancy between the old and new API concerning these two fields. In the old API they are byte values and converted to bits when calculating the final tag, but in the new API they need to be bit values for every KMA call. However, when thinking about this, only the final call to KMA would actually use the two values. 

So it might be possible to have byte values in the CPACF parm block until the last call. But I would leave it as is for now. 

At the end, the division by 8 in check_gcm_kma_parms would then be ok. And check_gcm_kma_parms is only called from the new API.

Reply:
Correction: the discrepancy is also and mainly between the simulation path and KMA path in the new API. And yes, unfortunately in the simulation case we have byte values in the ctx while the KMA path has bit values. So the new check_gcm_kma_parms routine cannot be implemented in this way. Thanks for seeing this!
I'm removing the last commit.